interfaces/builtin/serial_port.go

```diff
@@ -111,21 +174,47 @@ func (iface *SerialPortInterface) PermanentPlugSnippet(plug *interfaces.Plug, se
 func (iface *SerialPortInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
 	switch securitySystem {
 	case interfaces.SecurityAppArmor:
-		return []byte(fmt.Sprintf("%s rwk,\n", iface.path(slot))), nil
-	case interfaces.SecuritySecComp, interfaces.SecurityDBus, interfaces.SecurityUDev, interfaces.SecurityMount:
+		if len(slot.Attrs) == 1 {
+			// Path to fixed device node (no udev tagging)
+			path, pathOk := slot.Attrs["path"].(string)
+			if !pathOk {
+				return nil, nil
+			}
+			cleanedPath := filepath.Clean(path)
+			return []byte(fmt.Sprintf("%s rw,\n", cleanedPath)), nil
+		}
+
+		if len(slot.Attrs) == 3 {
+			// Wildcarded apparmor snippet as the cgroup will restrict down to the
+			// specific device
+			return []byte("/dev/serial-port-* rw,\n"), nil
+		}
```

Comment:
This is a symlink and needs to be resolved for apparmor to allow the access. However, I don't think we want to do this at all since resolving the symlink means we'll get a device in /dev whose name will change. We want to rely on the udev tagging and the device cgroups for this. Instead, this should be an apparmor glob that matches serialDeviceNodePattern. Specifically, use this instead:

```
# This apparmor rule must match serialDeviceNodePattern
return []byte("/dev/tty[A-Z]{,[A-Z],[A-Z][A-Z]}[0-9]{,[0-9],[0-9][0-9]} rw,\n"), nil
```

Unfortunately AARE is not as expressive as regex, but this should be equivalent for the current serialDeviceNodePattern. This could be simplified with `/dev/tty[A-Z]*[0-9]* rw,` but I prefer the exact match with serialDeviceNodePattern.